src/py21cmfast_tools/lc2ps.py

```diff
@@ -264,23 +351,55 @@ def log_bin(ps, kperp, kpar, bins=None):
         The number of bins or the bin edges to use for binning the kpar axis.
         If None, produces 16 bins logarithmically spaced between
         the minimum and maximum `kpar` supplied.
+    interp : str, optional
+        If 'linear', use linear interpolation to calculate the PS at the specified
+        kpar bins.
 
     """
     if bins is None:
-        bins = np.logspace(np.log10(kpar[0]), np.log10(kpar[-1]), 17)
+        if kpar_log:
+            bins = np.logspace(np.log10(kpar[0]), np.log10(kpar[-1]), 17)
+        else:
+            bins = np.linspace(kpar[0], kpar[-1], 17)
```

Comment:
why 17? should this not be something that's calculated based on the number of kpar?

Reply:
Good question, next question. I only added the if-else query for the linear k_par axis.

Reply:
Yeah, I would see what @DanielaBreitman says about this

Reply:
Sorry I missed these!! 17 is just a default number I put for now. It should be a tunable parameter. I'll add that.